services/scanner/agentpool/poolagent/agent.go

```diff
@@ -219,46 +227,90 @@ func (agent *Agent) SetClient(agentClient clients.AgentClient) {
 // from request channels.
 func (agent *Agent) StartProcessing() {
 	agent.initWait.Add(1)
-	go agent.initialize()
 
+	go func() {
+		err := agent.initWorker(agent.ctx)
```

Comment:
The chain of calls finally reach `agent.initialize()` and that seems to do all the logging. I think
```go
go agent.initWorker()
```
is sufficient here instead of anonymous `go func() { ... }`.

Could we also rename `initWorker()` to `startInitWorker()`? Then it's easier to understand that it's not initializing a worker but starting a worker to initialize (the bot :slightly_smiling_face: ).

Reply:
i think best way to do this it make both `initialize` and `initWorker` only return the errors, and log the errors at the end. I rather have all method return errors all the way to the highest call, rather than implicitly log errors inside the method

Reply:
If that makes sense, maybe let's go with that change. The downside is that there could be cases where we don't want to return an error (to not have a side effect in err checks) but still want to provide visibility to the error. And in that case, we would be both returning errs and logging.